chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,162 @@
+use near_metrics::{try_create_gauge_vec, try_create_int_gauge};
+use prometheus::{GaugeVec, IntGauge};
+use std::collections::HashMap;
+
+#[derive(Default)]
+/// Creates prometheus metrics on-demand for exporting RocksDB statistics.
+pub(crate) struct RocksDBMetrics {
+    // Contains counters and sums, which are integer statistics in RocksDB.
+    int_gauges: HashMap<String, IntGauge>,
+    // Contains floating point statistics, such as quantiles of timings.
+    gauges: HashMap<String, GaugeVec>,
```

Comment:
I wonder if these hash maps should be static singletons? my understanding that it’s only possible to register one counter with a specific name per process (prometheus register thing), so the current code looks like a lie: it seems that one can Crete independent instances of this struct, but this isn’t the case.

Reply:
Done